Source/API/galapagos.h

```diff
@@ -1,17 +1,39 @@
 #ifndef _GALAPAGOS_H_
 #define _GALAPAGOS_H_
 
+#include <functional>
+
 #include "galapagos_metadata.h"
+#include "selection_algorithm.h"
+#include "termination_condition.h"
 #include "population.h"
 #include "creature.h"
 #include "chromosome.h"
+#include "crossover.h"
+#include "mutation.h"
 
 #ifdef WIN32
 #define GALAPAGOS_API extern "C" __declspec(dllexport)
 #else
 #define GALAPAGOS_API extern "C" __attribute__((visibility("default")))
 #endif
 
+/****************************
+*****Galapagos Bootstrap*****
+****************************/
```

Comment:
Why do you need to pass the Galapagos object type in addition to the metadata? Like why can't I just do 
`typedef std::function<bool(mutation_metadata* metadata)> try_create_mutation_t;`?

Reply:
look at your signature. bool(mutation_metadata*) is the signature for a function that returns a bool and takes mutation_metadata* as argument. So your passing in the metadata to construct against and you are getting a return value indicating if the registered constructor (try_create_mutation_t) knows how to build an object from the metadata but callers of your construct (the genetic_factory) have now way of getting the pointer to the constructed object. If we add mutation* as the second argument then the caller has a way to get the pointer to the constructed object because the registered try_create assigned a value to the mutation*.